### PR TITLE
BBB for signal

### PR DESCRIPTION
**Pull request by scasasso**
see title

### CODE REVIEW

File: python/ShapeTools.py [hunk 1]
```diff
@@ -613,12 +613,10 @@ def optimizeExistingTemplates(self,pdf):
...
         return pdf
-    def createBBLiteVars(self, b,signal=False):
+
+    def createBBLiteVars(self, b, thr):
         print 'Doing bb-lite for bin ' + b
-        if signal:
-            procs = [p for p in self.DC.exp[b].keys() if self.DC.exp[b][p] != 0 and self.physics.getYieldScale(b,p) != 0 and self.DC.isSignal[p]]
-        else:
-            procs = [p for p in self.DC.exp[b].keys() if self.DC.exp[b][p] != 0 and self.physics.getYieldScale(b,p) != 0 and not self.DC.isSignal[p]]
+        procs = [p for p in self.DC.exp[b].keys() if self.DC.exp[b][p] != 0 and self.physics.getYieldScale(b,p) != 0 and not self.DC.isSignal[p]]
```
Comment (mcitron) [inline python/ShapeTools.py]: will this remove 0 bins?

File: python/ShapeTools.py [hunk 2]
```diff
@@ -635,22 +633,54 @@ def createBBLiteVars(self, b,signal=False):
...
+                binVarList.add(self.out.var(binvar))
+                binScaleList.add(self.out.function(binvar+'_Scale'))
+        binVarList.Print("v")
+        binScaleList.Print("v")
+        return (binVarList, binScaleList)
+
+
+    def createBBLiteVarsSig(self, b, thr):
+        print 'Doing bb-lite for signal in bin ' + b
+        procs = [p for p in self.DC.exp[b].keys() if self.DC.exp[b][p] != 0 and self.physics.getYieldScale(b,p) != 0 and self.DC.isSignal[p]]
```
Comment (mcitron) [inline python/ShapeTools.py]: will this remove zero?

**Comment by scasasso** [comment]
@mcitron updated as discussed

**Comment by scasasso** [comment]
@mcitron done, if you're happy merge
don't forget to pull the branch in the HCTBASE:
/vols/cms03/mc3909/HCT/CMSSW_7_1_5

**Pull request merged**